include/toolchain/gcc.h

```diff
@@ -179,6 +179,11 @@ do {                                                                    \
 #ifndef __packed
 #define __packed        __attribute__((__packed__))
 #endif
+
+#ifndef __PACKED_STRUCT
```

Comment:
This should be `__packed_struct`

Reply:
@carlescufi, the problem is that the baseline changes are leveraging the CMSIS headers and I think there are cases where the header file is not included so the author was trying to capture that with this particular change.

Reply:
CMSIS is ARM specific, so defining the GCC header based on that is a bit backwards.

Reply:
I get that CMSIS is ARM but this particular set of defines are agnostic. Can we consider adopting this template for toolchain specific support in the code base?

Reply:
The suggestion for having it lowercase was because that is how the other macros for this are defined, for example `__packed`
Is having it lowercase causing problems?

@carlescufi Any follow up here?

Reply:
I get why this is being asked for. I'm suggesting though that we consider adopting the CMSIS way of dealing with this as they have solved this problem for a larger set of compilers than we currently support.

Reply:
I don't get it, what is the point of using upper case? That CMSIS uses it? Not sure how this is relevant. This is Zephyr code in a Zephyr header that relies in the toolchain header files. If you want to use those headers outside of Zephyr you will anyway need to make other changes, since they contain Zephyr-specific includes, macros and even data structures. So I see no justification for deviating from the lower-case standard set in `gcc.h`.

Reply:
Agreed with @carlescufi here.